src/main/kotlin/org/dif/crypto/key/RecipientKeySelector.kt

```diff
@@ -1,24 +1,54 @@
 package org.dif.crypto.key
 
+import com.nimbusds.jose.jwk.Curve
 import org.dif.diddoc.DIDDocResolver
 import org.dif.exceptions.DIDDocException
 import org.dif.exceptions.DIDDocNotResolvedException
+import org.dif.exceptions.IncompatibleCryptoException
+import org.dif.exceptions.SecretNotFoundException
 import org.dif.secret.SecretResolver
 import org.dif.utils.divideDIDFragment
 import org.dif.utils.isDIDFragment
 
 class RecipientKeySelector(private val didDocResolver: DIDDocResolver, private val secretResolver: SecretResolver) {
-    fun verifyKey(selector: String): Key = Key.wrapVerificationMethod(
+    fun findVerificationKey(signFrom: String): Key = Key.wrapVerificationMethod(
         let {
-            val (did) = divideDIDFragment(selector)
-            val didDoc = didDocResolver.resolve(did).orElseThrow { throw DIDDocNotResolvedException(did) }
+            check(isDIDFragment(signFrom)) { "'DID URL' was excepted" }
```

Comment:
I think a more detailed message is needed (something like `'DID URL' is expected as a signature verification key. Got: ${signFrom}`)